### PR TITLE
dlgBarcharts work

### PR DESCRIPTION
**Pull request by maxwellfundi**

### CODE REVIEW

File: instat/dlgBarAndPieChart.vb [hunk 1]
```diff
@@ -99,16 +104,16 @@ Public Class dlgBarAndPieChart
...
+
         If rdoBarChart.Checked = True Then
             clsRgeom_barchart.RemoveParameterByName("width")
             ucrBase.clsRsyntax.RemoveOperatorParameter("polar")
             cmdBarChartOptions.Visible = True
             cmdPieChartOptions.Visible = False
             ucrSecondReceiver.Visible = True
             lblSecondFactor.Visible = True
-        ElseIf rdoPieChart.Checked = True Then
+        ElseIf rdoPieChart.Checked = True
```
Comment (dannyparsons) [inline instat/dlgBarAndPieChart.vb]: Isn't this going to give an error now?

File: instat/sdgPieChartOptions.vb [hunk 2]
```diff
@@ -14,8 +14,25 @@
...
+    Public clsPieChartFunction As RFunction
+    Public Sub SetPieChartFunction(clsPieChartFunc As RFunction)
+        clsPieChartFunction = clsPieChartFunc
+    End Sub
+
+    Public Sub SetDefaults()
+        nudStartPoint.Value = "0"
+        chkClockwise.Checked = True
+    End Sub
+    Private Sub nudStartPoint_Leave(sender As Object, e As EventArgs) Handles nudStartPoint.Leave
```
Comment (dannyparsons) [inline instat/sdgPieChartOptions.vb]: Use the textChanged event for numeric up downs. We try to avoid using Leave if possible. And also check that Text <> "", and otherwise remove the parameter. @Lunalo has an example of a working nud on one of his dialogs if you need to see an example.

File: instat/dlgBarAndPieChart.vb (hunk 1 above)
Comment (maxwellfundi) [inline instat/dlgBarAndPieChart.vb]: @dannyparsons  i tested it and it has no error. Why do you think will give an error?

Comment (dannyparsons) [inline instat/dlgBarAndPieChart.vb]: Without "Then" it should give an error. Why did you remove it anyway?

Comment (maxwellfundi) [inline instat/dlgBarAndPieChart.vb]: that might have been an error. AM fixing it now

**Comment by maxwellfundi** [comment]
@dannyparsons i did this already

File: instat/sdgPieChartOptions.vb [hunk 3]
```diff
@@ -24,8 +24,11 @@ Public Class sdgPieChartOptions
         nudStartPoint.Value = "0"
         chkClockwise.Checked = True
     End Sub
-    Private Sub nudStartPoint_Leave(sender As Object, e As EventArgs) Handles nudStartPoint.Leave
-        clsPieChartFunction.AddParameter("start", nudStartPoint.Value)
+    Private Sub nudStartPoint_TextChanged(sender As Object, e As EventArgs) Handles nudStartPoint.TextChanged
+        If nudStartPoint.Value <> "" Then
+            clsPieChartFunction.AddParameter("start", nudStartPoint.Value)
+            clsPieChartFunction.RemoveParameterByName("start")
```
Comment (dannyparsons) [inline instat/sdgPieChartOptions.vb]: You just forgot the Else in this part.
